SourceX/storm/storm.cpp

```diff
@@ -624,6 +632,10 @@ BOOL SVidPlayBegin(char *filename, int a2, int a3, int a4, int a5, int flags, HA
 #else
 	if (SDL_SetSurfacePalette(SVidSurface, SVidPalette) <= -1) {
 #endif
+
+#if defined(USE_SDL1) && D_BPP == 8
+		SDL_SetColors(surface, SVidPalette->colors, 0, SVidPalette->ncolors);

```

Comment:
Is the call to `SDL_SetPalette` needed when the main `surface` is itself and index texture? Also maybe this line should be moved into the previous `#ifdef USE_SDL1` since it appears to be part of the same action.

Reply:
Yes, both setpalette and setcolors are needed. I've tested this, and this is what works when the main window surface is 8bit.

Reply:
And yes, it can probably be moved up. Except that the statement above is an "if not" statement, and I wanted it to set the colors after the fact, so to assure it had not failed to set the palette of the video surface.